src/rt/aaA.d

```diff
@@ -699,16 +699,25 @@ extern (C) pure nothrow @nogc
 
     void* _aaRangeFrontKey(Range r)
     {
+        if (r.idx >= r.dim)
```

Comment:
Marking the function `@safe` enables bounds checking on the buckets-access, so I think the test here is unnecessary. A range error is probably better than an access violation later.

Interestingly, x64 code elides the bounds check if this condition exists, but keeps both for 32-bit code.